File: roboco/api/schemas/company_goals.py
"""Company-goals API schemas — the CEO-owned company charter."""

from typing import Any

from pydantic import BaseModel, Field


class CompanyGoalsResponse(BaseModel):
    """The company charter as returned to any authenticated agent."""

    north_star: str
    objectives: list[dict[str, Any]]
    constraints: list[str]
    operating_policy: dict[str, Any]
    updated_at: str | None = None
    updated_by: str | None = None


class CompanyGoalsUpdate(BaseModel):
    """Partial update to the charter (CEO-only); only provided fields change."""

    north_star: str | None = Field(default=None)
    objectives: list[dict[str, Any]] | None = Field(default=None)
    constraints: list[str] | None = Field(default=None)
    operating_policy: dict[str, Any] | None = Field(default=None)

File: panel/src/lib/api/company-goals.ts
import api from "./client";

export interface CompanyGoals {
  north_star: string;
  objectives: Record<string, unknown>[];
  constraints: string[];
  operating_policy: Record<string, unknown>;
  updated_at?: string | null;
  updated_by?: string | null;
}

export type CompanyGoalsUpdate = Partial<
  Pick<
    CompanyGoals,
    "north_star" | "objectives" | "constraints" | "operating_policy"
  >
>;

export const companyGoalsApi = {
  // GET /api/company-goals — the charter (any authenticated agent).
  get: async (): Promise<CompanyGoals> => {
    const { data } = await api.get<CompanyGoals>("/company-goals");
    return data;
  },
  // PUT /api/company-goals — CEO-only; partial update, returns the full charter.
  update: async (update: CompanyGoalsUpdate): Promise<CompanyGoals> => {
    const { data } = await api.put<CompanyGoals>("/company-goals", update);
    return data;
  },
};
